src/plugins/chart_expressions/expression_xy/public/helpers/annotations.tsx

```diff
@@ -31,7 +39,9 @@ export const getLinesCausedPaddings = (
     if (!config) {
       return;
     }
-    const { axisMode, icon, iconPosition, textVisibility } = config;
+    const { axisMode, icon, textVisibility } = config;
+    const iconPosition = (config as PartialExtendedYConfig).iconPosition ?? undefined;
```

Comment:
how about changing it to: `const iconPosition = 'iconPosition' in config ? config.iconPosition : undefined;` to avoid type assertion?

Reply:
I dislike the construction `'tokens' in obj`. It will be very difficult to detect problems with them in the future. But I also agree that we should wrap up the `as type` statement into the guard. Performed this change.